one-off/nuke-workitem-embedded-images.sql

```diff
@@ -0,0 +1,174 @@
+create or alter function [dbo].[OidSplit] (
+    @oid varchar(200)
+)
+returns table
+as
+return
+(
+    select 
+        substring(@oid, 1, charindex(':', @oid) - 1) COLLATE Latin1_General_BIN as AssetType,
+        cast(substring(@oid, charindex(':', @oid) + 1, len(@oid)) as int) as ID
+)
+GO
+
+create or alter function [dbo].[ExtractHashesFromUrls]
+(
+    @richtextcol nvarchar(max)
+)
+returns table
+as
+return
+(
+    with recursivecte as (
+        select
+            charindex('<img', @richtextcol) as startpos,
+            charindex('>', @richtextcol, charindex('<img', @richtextcol)) as endpos,
+            1 as rownumber
+        where charindex('<img', @richtextcol) > 0
+
+        union all
+
+        select
+            charindex('<img', @richtextcol, endpos) as startpos,
+            charindex('>', @richtextcol, charindex('<img', @richtextcol, endpos)) as endpos,
+            rownumber + 1
+        from
+            recursivecte
+        where
+            charindex('<img', @richtextcol, endpos) > 0
+    )
+    select
+        substring(
+            @richtextcol,
+            startpos,
+            endpos - startpos + 1
+        ) as imgelem,
+        convert(binary(20), '0x'+ substring(
+            @richtextcol,
+            charindex('downloadblob.img/', @richtextcol, startpos) + len('downloadblob.img/'),
+            charindex('"', @richtextcol, charindex('downloadblob.img/', @richtextcol, startpos) + len('downloadblob.img/')) - charindex('downloadblob.img/', @richtextcol, startpos) - len('downloadblob.img/')
+        ),1) as hash
+    from
+        recursivecte
+    where
+        charindex('downloadblob.img/', @richtextcol, startpos) > 0
```

Comment:
i think you can just look for `downloadblob.img/` as the starting marker, and the immediately following `"` as the trailing marker for the hash, without bothering with it being in an `<img ... >`